admin/src/bench/mod.rs

```diff
@@ -231,16 +230,38 @@ fn writer<D: BenchDb>(
 		commit.push((KEY_RESTART, Some((n as u64).to_be_bytes().to_vec())));
 
 		db.commit(commit.drain(..));
-		COMMITS.fetch_add(1, Ordering::Release);
 		commit.clear();
+
+		if n >= start_commit + args.commits || shutdown.load(Ordering::Relaxed) {
+			break
+		}
 	}
-	commit.clear();
 }
 
-fn reader<D: BenchDb>(_db: Arc<D>, shutdown: Arc<AtomicBool>) {
-	// Query a random  key
+fn reader<D: BenchDb>(
+	db: Arc<D>,
+	pool: Arc<SizePool>,
+	seed: u64,
+	index: u64,
+	shutdown: Arc<AtomicBool>,
+) {
+	// Query random keys while writing
+	let mut rng = rand::rngs::SmallRng::seed_from_u64(seed + index);
 	while !shutdown.load(Ordering::Relaxed) {
-		thread::sleep(std::time::Duration::from_millis(500));
+		let commits = COMMITS.load(Ordering::Relaxed) as u64;
+		if commits == 0 {
```

Comment:
start_commit rather than 0? Actually since start_commit is restart index, there should be content so probably good like this.

Reply:
`COMMITS` is the number of commits in the database. It should be possible to query any of them.

Reply:
yes thought so, certainly no use to target specifically the latest ones.